controllers/commonservice_controller.go

```diff
@@ -57,40 +64,143 @@ func (r *CommonServiceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
+		return ctrl.Result{}, nil
 	}
 
-	if instance.Spec.Size == "medium" {
-		newConfigs = deepMerge(instance.Spec.Services, size.Medium)
+	switch instance.Object["spec"].(map[string]interface{})["size"] {
+	case "small":
+		if instance.Object["spec"].(map[string]interface{})["services"] == nil {
+			newConfigs = deepMerge(newConfigs, size.Small)
+		} else {
+			newConfigs = deepMerge(instance.Object["spec"].(map[string]interface{})["services"].([]interface{}), size.Small)
+		}
+	case "medium":
+		if instance.Object["spec"].(map[string]interface{})["services"] == nil {
+			newConfigs = deepMerge(newConfigs, size.Medium)
+		} else {
+			newConfigs = deepMerge(instance.Object["spec"].(map[string]interface{})["services"].([]interface{}), size.Medium)
+		}
+	case "large":
+		if instance.Object["spec"].(map[string]interface{})["services"] == nil {
+			newConfigs = deepMerge(newConfigs, size.Large)
+		} else {
+			newConfigs = deepMerge(instance.Object["spec"].(map[string]interface{})["services"].([]interface{}), size.Large)
+		}
 	}
 
-	if instance.Spec.Size == "large" {
-		newConfigs = deepMerge(instance.Spec.Services, size.Large)
+	err = r.updateOpcon(ctx, newConfigs)
+
+	if err != nil {
+		return ctrl.Result{}, err
 	}
+	return ctrl.Result{}, nil
+}
 
-	if isConfigChanged() {
-		err := patchOperandConfig(newConfigs)
-		if err != nil {
-			return ctrl.Result{}, err
+func (r *CommonServiceReconciler) updateOpcon(ctx context.Context, newConfigs []interface{}) error {
+	opcon := &unstructured.Unstructured{}
+	opcon.SetGroupVersionKind(schema.GroupVersionKind{Group: "operator.ibm.com", Kind: "OperandConfig", Version: "v1alpha1"})
+	err := r.Client.Get(ctx, types.NamespacedName{
+		Name:      "common-service",
+		Namespace: "ibm-common-services",
+	}, opcon)
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+	services := opcon.Object["spec"].(map[string]interface{})["services"].([]interface{})
+	for _, service := range services {
+		for _, size := range newConfigs {
+			if service.(map[string]interface{})["name"].(string) == size.(map[string]interface{})["name"].(string) {
+				for cr, spec := range service.(map[string]interface{})["spec"].(map[string]interface{}) {
+					if size.(map[string]interface{})["spec"].(map[string]interface{})[cr] == nil {
+						continue
+					}
+					service.(map[string]interface{})["spec"].(map[string]interface{})[cr] = mergeConfig(spec.(map[string]interface{}), size.(map[string]interface{})["spec"].(map[string]interface{})[cr].(map[string]interface{}))
+				}
+			}
 		}
 	}
 
-	return ctrl.Result{}, nil
-}
+	opcon.Object["spec"].(map[string]interface{})["services"] = services
+
+	err = r.Client.Update(ctx, opcon)
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
 
-func deepMerge(src []apiv3.ServiceConfig, dest string) []apiv3.ServiceConfig {
 	return nil
 }
 
-func patchOperandConfig(newConfigs []apiv3.ServiceConfig) error {
-	return nil
+func deepMerge(src []interface{}, dest string) []interface{} {
```

Comment:
/lgtm cancel
According to the bot test, Should the type of parameter `dest` be one of services instead of string?